### PR TITLE
feat(saved-searches): Add a saved searches sidebar and header button

### PR DESCRIPTION
**Pull request by malwilley**
Visible on the sentry-test org

image

Future work:

- Overflow menu on saved search items
- Add the saved search button to the sidebar
- Consider using localStorage for sidebar open state
- Better responsive styling
- Remove saved searches dropdown

### CODE REVIEW

File: static/app/views/issueList/savedIssueSearches.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  it('displays saved searches with correct text and in correct sections', function () {
+    const {container} = render(<SavedIssueSearches {...defaultProps} />);
+
+    expect(container).toSnapshot();
+  });
+
+  it('can select a saved search', function () {
+    render(<SavedIssueSearches {...defaultProps} />);
+
+    userEvent.click(screen.getByText('Assigned to Me'));
```
[the remark this answers is not in the document]
Comment (malwilley) [reply static/app/views/issueList/savedIssueSearches.spec.tsx]: Yeah you're right, added an aria-label (so the button is labeled by the name and not the query) and changed to getByRole

File: static/app/views/issueList/header.tsx [hunk 2]
```diff
@@ -89,6 +93,15 @@ function IssueListHeader({
       <Layout.HeaderActions>
         <ButtonBar gap={1}>
           <IssueListSetAsDefault {...{sort, query, savedSearch, organization}} />
+          {organization.features.includes('issue-list-saved-searches-v2') && (
+            <Button
+              size="sm"
+              icon={<IconStar size="sm" isSolid={isSavedSearchesOpen} />}
+              onClick={() => onToggleSavedSearches(!isSavedSearchesOpen)}
+            >
+              Saved Searches
```
Comment (evanpurkhiser) [inline static/app/views/issueList/header.tsx:102]:
```suggestion
              {t('Saved Searches')}
```

File: static/app/views/issueList/savedIssueSearches.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      <StyledSidebar>
+        <LoadingIndicator />
+      </StyledSidebar>
+    );
+  }
+
+  const [recommendedSavedSearches, orgSavedSearches] = partition(
+    savedSearches,
+    item => item.isGlobal
+  );
```
Comment (evanpurkhiser) [inline static/app/views/issueList/savedIssueSearches.tsx:65]: We may also want to filter out `isPinned` as well?

Comment (evanpurkhiser) [reply static/app/views/issueList/savedIssueSearches.tsx:65]: pinned save searches do come back from the API

File: static/app/views/issueList/savedIssueSearches.tsx [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+  return (
+    <StyledSidebar>
+      {orgSavedSearches.length > 0 && (
+        <Fragment>
+          <Heading>Saved Searches</Heading>
```
Comment (evanpurkhiser) [inline static/app/views/issueList/savedIssueSearches.tsx:71]:
```suggestion
          <Heading>{t('Saved Searches')}</Heading>
```

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/views/issueList/savedIssueSearches.tsx (hunk 3 above)
Comment (malwilley) [reply static/app/views/issueList/savedIssueSearches.tsx:65]: Good catch! Will filter those out

**Pull request merged**